addons/sale_management/models/sale_order.py

```diff
@@ -158,20 +158,6 @@ def action_confirm(self):
                 self.sale_order_template_id.mail_template_id.send_mail(order.id)
         return res
 
-    def get_access_action(self, access_uid=None):
-        """ Instead of the classic form view, redirect to the online quote if it exists. """
-        self.ensure_one()
-        user = access_uid and self.env['res.users'].sudo().browse(access_uid) or self.env.user
-
-        if not self.sale_order_template_id or (not user.share and not self.env.context.get('force_website')):
-            return super(SaleOrder, self).get_access_action(access_uid)
-        return {
-            'type': 'ir.actions.act_url',
-            'url': self.get_portal_url(),
```

Comment:
I don't know those mail flows well, but if the `force_website` context is given, the old code would return the url with the access_token (through `get_portal_url`, whereas the `portal` `get_access_action` would return the `access_url` only, which doesn't include the access_token by default :thinking:.

Are you sure this won't be a problem and the links returned by some flows won't be forbidden for some users previously able to reach their SO?

Reply:
Looking at the call chain I think main use case do not change (did test some basic flows everything was working as intended (or probably intended)). Corner case may arise (sudo, not sudo, access_uid, ...). I think it is better to leak a bit less the access token and fix afterwards if necessary than the inverse move.

I think it is better to remove an override that do not handle access_uid properly and rely on standard code which is already complicated.

Reply:
Ok then, It's not even sure it will trigger problems, was just highlighting the change to be sure it was considered.
Anyway, if something must be fixed/modified, it's probably not in such a custom hacky override/way.

Reply:
Saw the notification by mail and checked out of curiosity.
Seems to be reverting https://github.com/odoo/odoo/commit/b49da62365e442f780c50fa6290edf07cf375ac4 but I am not too sure how to reproduce it or if even still needed.
You definitely know better, but just to be sure :)

Reply:
Thanks for pointing it out. Indeed seems to be monkey patch of old code when sale did not use portal but website quote did (with its "accept quote online" specific code that is now standard). Now everything is done directly in sale (accept, pay, ...), and website_quote has been globally merged with sale I think.